### PR TITLE
url encode logout url

### PR DESCRIPTION
**Pull request by brutif**

### CODE REVIEW

**Comment by brutif** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.openidconnect.server_fat.endpoint.end_session,com.ibm.ws.security.openidconnect.server_fat.endpoint.introspect,com.ibm.ws.security.openidconnect.server_fat.endpoint.revoke,com.ibm.ws.security.openidconnect.server_fat.endpoint.token,com.ibm.ws.security.openidconnect.server_fat.endpoint.userinfo,com.ibm.ws.security.openidconnect.server_fat.feature,com.ibm.ws.security.openidconnect.server_fat.granttype.jwt,com.ibm.ws.security.openidconnect.server_fat,com.ibm.ws.security.openidconnect.server_fat.jaxrs,com.ibm.ws.security.openidconnect.server_fat.config,com.ibm.ws.security.openidconnect.server_fat.jaxrs.config,com.ibm.ws.security.openidconnect.server_fat.endpoint.authorize,com.ibm.ws.security.openidconnect.server_fat.oauth,com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration,com.ibm.ws.security.openidconnect.server_fat.oidc,com.ibm.ws.security.openidconnect.server_fat.endpoint.coveragemap,com.ibm.ws.security.openidconnect.server_fat.saml,com.ibm.ws.security.openidconnect.server_fat.endpoint.discovery

File: dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/OAuth20EndpointServices.java [hunk 1]
```diff
@@ -474,6 +477,23 @@ public void logout(OAuth20Provider provider,
         }
     }
 
+    String URLEncodeParams(String UrlStr) {
+        String sep = "?";
+        String encodedURL = UrlStr;
+        int index = UrlStr.indexOf(sep);
+        if (index > -1) {
+            index += 3;
```
Comment (ayoho) [inline dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/OAuth20EndpointServices.java]: I'm not seeing where this `+3` comes into play; right now this falls under the category of a magic number. I could understand adding 1 to skip over the `?` character, but 3?

File: dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/web/OAuth20EndpointServicesTest.java [hunk 2]
```diff
@@ -183,6 +183,19 @@ public void testIntrospectExpired() {
         }
     }
 
+    @Test
+    public void testURLEncodeParams() {
+        OAuth20EndpointServices oas = new OAuth20EndpointServices();
+        String in = "https://www.somewhere.com?item=s巧";
+        String expected = "https://www.somewhere.com?item%3Ds%E5%B7%A7";
```
Comment (ayoho) [inline dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/web/OAuth20EndpointServicesTest.java]: I don't think is actually what we want - the `=` characters shouldn't get getting encoded unless they're actually part of a parameter name or value. You should be able to do an experiment to see the difference in behavior by going to these pages:
- https://github.com/OpenLiberty/open-liberty/pull/9706/files?w=1 (Loads the diff ignoring white space)
- https://github.com/OpenLiberty/open-liberty/pull/9706/files?w%3D1 (The encoded `=` character is considered part of the parameter name, so the browser will actually just end up dropping that parameter entirely and will *not* load the diff ignoring white space)

File: dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/OAuth20EndpointServices.java (hunk 1 above)
Comment (brutif) [reply dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/OAuth20EndpointServices.java]: fixed, thanks

File: dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/web/OAuth20EndpointServicesTest.java (hunk 2 above)
Comment (brutif) [reply dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/web/OAuth20EndpointServicesTest.java]: fixed, thanks

**Comment by brutif** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.openidconnect.server_fat.endpoint.end_session,com.ibm.ws.security.openidconnect.server_fat.endpoint.introspect,com.ibm.ws.security.openidconnect.server_fat.endpoint.revoke,com.ibm.ws.security.openidconnect.server_fat.endpoint.token,com.ibm.ws.security.openidconnect.server_fat.endpoint.userinfo,com.ibm.ws.security.openidconnect.server_fat.feature,com.ibm.ws.security.openidconnect.server_fat.granttype.jwt,com.ibm.ws.security.openidconnect.server_fat,com.ibm.ws.security.openidconnect.server_fat.jaxrs,com.ibm.ws.security.openidconnect.server_fat.config,com.ibm.ws.security.openidconnect.server_fat.jaxrs.config,com.ibm.ws.security.openidconnect.server_fat.endpoint.authorize,com.ibm.ws.security.openidconnect.server_fat.oauth,com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration,com.ibm.ws.security.openidconnect.server_fat.oidc,com.ibm.ws.security.openidconnect.server_fat.endpoint.coveragemap,com.ibm.ws.security.openidconnect.server_fat.saml,com.ibm.ws.security.openidconnect.server_fat.endpoint.discovery

**Comment by brutif** [comment]
looks like a lot of child builds died so lets try it again.

#build
#spawn.fullfat.buckets=com.ibm.ws.security.openidconnect.server_fat.endpoint.end_session,com.ibm.ws.security.openidconnect.server_fat.endpoint.introspect,com.ibm.ws.security.openidconnect.server_fat.endpoint.revoke,com.ibm.ws.security.openidconnect.server_fat.endpoint.token,com.ibm.ws.security.openidconnect.server_fat.endpoint.userinfo,com.ibm.ws.security.openidconnect.server_fat.feature,com.ibm.ws.security.openidconnect.server_fat.granttype.jwt,com.ibm.ws.security.openidconnect.server_fat,com.ibm.ws.security.openidconnect.server_fat.jaxrs,com.ibm.ws.security.openidconnect.server_fat.config,com.ibm.ws.security.openidconnect.server_fat.jaxrs.config,com.ibm.ws.security.openidconnect.server_fat.endpoint.authorize,com.ibm.ws.security.openidconnect.server_fat.oauth,com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration,com.ibm.ws.security.openidconnect.server_fat.oidc,com.ibm.ws.security.openidconnect.server_fat.endpoint.coveragemap,com.ibm.ws.security.openidconnect.server_fat.saml,com.ibm.ws.security.openidconnect.server_fat.endpoint.discovery

**Comment by brutif** [comment]
#run-libby-bot

**Pull request merged**